src/layers/legacy/medVtkInria/vtkImageView/vtkImageView2D.cxx

```diff
@@ -2081,31 +2091,28 @@ void vtkImageView2D::UpdateBounds (const double bounds[6], int layer, vtkMatrix4
 }
 
 //----------------------------------------------------------------------------
-void vtkImageView2D::RemoveDataSet (vtkPointSet *arg)
+void vtkImageView2D::RemoveDataSet(vtkPointSet *arg)
 {
-  this->Superclass::RemoveDataSet (arg);
-
-  for (std::list<vtkDataSet2DWidget*>::iterator it = this->DataSetWidgets.begin();
-      it != this->DataSetWidgets.end(); ++it)
-  {
-    if ((*it)->GetSource()==arg)
+    this->Superclass::RemoveDataSet(arg);
+    for (auto &dataSetWidget : this->DataSetWidgets)
     {
-      (*it)->Off();
-      (*it)->SetImageView (nullptr);
-      (*it)->Delete();
-      this->DataSetWidgets.erase (it);
-      this->Modified();
-      break;
+        if (dataSetWidget->GetSource() == arg)
+        {
+            dataSetWidget->Off();
+            dataSetWidget->SetImageView(nullptr);
+            dataSetWidget->Delete();
+            this->DataSetWidgets.remove(dataSetWidget);
+            this->Modified();
+            break;
+        }
     }
-  }
 }
 
 //----------------------------------------------------------------------------
 std::list<vtkDataSet2DWidget*>::iterator vtkImageView2D::FindDataSetWidget(vtkPointSet* arg)
 {
 
-    for (  std::list<vtkDataSet2DWidget*>::iterator it = this->DataSetWidgets.begin();
-            it != this->DataSetWidgets.end(); ++it)
+  for (std::list<vtkDataSet2DWidget*>::iterator it = this->DataSetWidgets.begin(); it != this->DataSetWidgets.end(); ++it)
```

Comment:
would it be possible to use a `for (aa : bb)` with the ` vtkDataSet2DWidget* widget` under?

Reply:
It's not possible because, for the moment, it return an iterator